server/src/main/java/org/candlepin/policy/js/pool/PoolHelper.java

```diff
@@ -143,18 +149,18 @@ private PoolHelper() {
 
                 String subscriptionId = pool.getSubscriptionId();
                 if (subscriptionId != null && !subscriptionId.isEmpty()) {
-                    consumerSpecificPool.setSourceSubscription(new SourceSubscription(subscriptionId,
-                        sourceEntitlements.get(pool.getId()).getId()));
+                    delayedPoolOp.createSourceSubscription(consumerSpecificPool, subscriptionId,
+                        sourceEntitlements.get(pool.getId()));
                 }
             }
-            poolsToCreate.add(consumerSpecificPool);
+            delayedPoolOp.addPoolToCreate(consumerSpecificPool);
         }
 
         if (CollectionUtils.isNotEmpty(poolsToUpdateFromStack)) {
-            poolManager.updatePoolsFromStack(consumer, poolsToUpdateFromStack);
+            poolManager.updatePoolsFromStackWithoutDeletingStack(consumer, poolsToUpdateFromStack, null);
```

Comment:
@wottop  because this method invocation intentionally does not delete stack derived pools, and hence there is no need for a delayed execution. I renamed the method to WithoutDeletingStack  to convey that information, but perhaps there is a better name needed If I got this question :)
Any suggestion?